src/extension.ts

```diff
@@ -171,41 +171,121 @@ class RunLineInEditor {
-                {
-                    this.commandRunningStatusBarItem.text = `${this.statusBarItemText} ${this.statusBarSpinner()}`;
-                }
-                else
-                {
-                    this.commandRunningStatusBarItem.text = `${this.statusBarItemText} [${this.runningCommandCount}] ${this.statusBarSpinner()}`;
+            this.runningCommandCount += 1;
+            const t0 = Date.now();
+            if (this.runningCommandCount === 1)
+            {
+                this.statusBarItemText = `Azure CLI: Waiting for response`;
+                this.statusBarUpdateInterval = setInterval(() => {
+                    if (this.runningCommandCount === 1)
+                    {
+                        this.commandRunningStatusBarItem.text = `${this.statusBarItemText} ${this.statusBarSpinner()}`;
+                    }
+                    else
+                    {
+                        this.commandRunningStatusBarItem.text = `${this.statusBarItemText} [${this.runningCommandCount}] ${this.statusBarSpinner()}`;
+                    }
+                }, 50);
+            }
+            this.commandRunningStatusBarItem.show();
+            clearTimeout(this.hideStatusBarItemTimeout);
+
+            this.parsedResult = undefined;
+            this.query = undefined; // TODO
+            return this.findResultDocument()
+                .then(document => window.showTextDocument(document, ViewColumn.Two, true))
+                .then(target => replaceContent(target, JSON.stringify({ 'Running command': command }) + '\n')
+                    .then(() => exec(command))
+                    .then(({ stdout }) => stdout, ({ stdout, stderr }) => JSON.stringify({ stderr, stdout }, null, '    '))
+                    .then(content => replaceContent(target, content)
+                            .then(() => this.parsedResult = JSON.parse(content))
+                            .then(undefined, err => {})
+                    )
+                    .then(() => this.commandFinished(t0))
+                )
+                .then(undefined, console.error);
+        }
+    }
+
+    private GetSelectedCommand(source: TextEditor)
+    {
+        if (source.selection.isEmpty)
+        {
+            var lineNumber = source.selection.active.line;
+            if (source.document.lineAt(lineNumber).text.length === 0)
+            {
+                window.showInformationMessage<any>("Please put the cursor on a line that contains a command.");
+                return "";
+            }
+            
+            // find the start of the command (if necessary)
+            while(!source.document.lineAt(lineNumber).text.trim().toLowerCase().startsWith("az"))
+            {
+                lineNumber--;
+            }
+
+            var command = this.StripComments(source.document.lineAt(lineNumber).text);
+
+            // using backtick (`) as continuation character
+            while (command.trim().endsWith("`"))
```

Comment:
'`' is problematic because it can be part of a Bash command. The continuation character depends on the shell. A good approximation might be to use backslash on Mac and Linux and backtick on Windows.

Reply:
I forget why I decided on ` as the continuation character; probably because it is rarely used. Given that this extension only applies to .azcli scrapbooks do shell/OS considerations come into play or can we just decide on a char that is the continuation char for .azcli files?

Reply:
Isn't that the continuation character for PowerShell?

Reply:
I think so; it's been a while since I did any PS. Do you think shell/OS considerations are applicable in a .azcli scrapbook?

Reply:
@chrmarti  I've pulled the continuation character into a class variable to make it easy to change. What would you think about putting it in a user preference?

Reply:
Sounds good. I'd also make the default depend on the platform (use backtick on Windows and hash on the other platforms).

Reply:
Currently # is used to denote a comment; did you mean backslash (for the non Windows continuation character)?

Reply:
I meant backslash. :)